XX-eltoo-transactions.md

```diff
@@ -129,10 +129,10 @@ where EXPR_UPDATE(n) =
 
 and where EXPR_SETTLE(n) =
 
-`CovSig(n) 1_G OP_CHECKSIG`
+`CovSig(n) 1 OP_CHECKSIG`
 
 where `CovSig(n)` is the SIGHASH_ALL|ANYPREVOUTANYSCRIPT signature of the corresponding settlement transaction with a
-locktime of `n`, and `1_G` the 33-byte BIP118 public key matching the secp256k1 generator `G`.
+locktime of `n` signed by `aggregated_key` as per BIP-musig2/BIP-118.
```

Comment:
You need a signature from your counterparty for the update tx, might as well get one for the settlement at the same time; that way you save 33 bytes of witness data onchain...

Reply:
As ariard noted elsewhere(which is why im now suggesting issues going forward), if you send your partial sigs for both update and settle txns, your counterparty can go to chain with the update tx and you then have a hostage situation for getting settlement out.

To fix this, I think, you either need another RTT to receive settlement full signature first before sending update tx partial sig, or you use this strategy.

We could also redefine an OP_SUCCESSX to push this value to stack, if we think it's going to be used a lot, as Jeremy noted in some email.

Reply:
a) don't need an OP_SUCCESSX, could have "1" be bip-118 internal pubkey, and "-1" be bip-118 sign by the generator

b) the hostage scenario can't exist -- you need to know the settlement signature in order to finish creating the update tx in the first place, and you can't sign the update tx before you've finished creating it. that definitely adds a round-trip though :(

Reply:
oh right, I was thinking you meant a traditional pre-signed transaction, not sticking the musig signature in the output script. Same issue with RTT IIUC